### PR TITLE
CORDA-578 Make CordaX500Name usable as web query param

### PR DESCRIPTION
**Pull request by andr3ej**
there's probably a less invasive way to do this, but i reckon it would be 10x as much code

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt [hunk 1]
```diff
@@ -108,7 +107,11 @@ data class CordaX500Name(val commonName: String?,
         }
 
         @JvmStatic
-        fun parse(name: String) : CordaX500Name = build(X500Principal(name))
+        fun parse(name: String): CordaX500Name = build(X500Principal(name))
+
+        /** For JAX-RS integration only, prefer [parse] in regular code. */
+        @JvmStatic
+        fun valueOf(name: String) = parse(name)
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: Why can't we use `parse`?

Comment (andr3ej) [reply core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: In cordapp-tutorial there is some web magic that converts a query arg into an object, and one of the ways to satisfy it is by adding valueOf. Currently the webservers crash on startup. See https://stackoverflow.com/questions/29676563/jersey-modelvalidationexception-no-injection-source-found/29676677#29676677

Comment (andr3ej) [reply core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: I guess X500Name worked because it has a String constructor

Comment (m4ksio) [reply core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: Other approach would be to change corda-tutorial to use JSON objects rather then query param

Comment (shamsasari) [reply core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: `CordaX500Name` is core public API and `parse` is a better name than `valueOf`. The change should occur in the tutorial and not here.

Comment (shamsasari) [reply core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: To elaborate more, `valueOf` isn't totally a bad name, but to add this method solely to support a third-party API isn't the right direction.

Comment (josecoll) [reply core/src/main/kotlin/net/corda/core/identity/CordaX500Name.kt]: I agree - we should not be adding specific API calls to support web applications.

**Comment by andr3ej** [comment]
Neither review is satisfactory so here's mine: It's not enough that the new method is clearly marked as not API - if a user were to use that method via a web framework maybe by accident, then we're not going to be able to justify changing/removing it.

**Comment by mikehearn** [comment]
Huh - this looked OK to me! Why are we turning down useful convenience stuff like this? It's such a trivial API that it can't really be "wrong", the only thing that might be a concern is security. There's no real cost to having aliases like this if it helps simplify things elsewhere.

**Comment by m4ksio** [comment]
This change was essentially done to support conversion for JAX WS [QueryParam](http://docs.oracle.com/javaee/7/api/javax/ws/rs/QueryParam.html) - I doubt we should add some 'magic' methods to support some unrelated APIs.

**Comment by vkolomeyko** [comment]
From Javadoc of QueryParam:

```
The type T of the annotated parameter, field or property must either:
1. Be a primitive type
2. Have a constructor that accepts a single String argument
3. Have a static method named valueOf or fromString that accepts a single String argument (see, for example, Integer.valueOf(String))
4. Have a registered implementation of javax.ws.rs.ext.ParamConverterProvider JAX-RS extension SPI that returns a javax.ws.rs.ext.ParamConverter instance capable of a "from string" conversion for the type.
...
```

If people object to this simple change, perhaps we should do it via `ParamConverter` ?

**Pull request closed without merging**